nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -653,9 +653,28 @@ def _register_reduced_input_impl(self, x: TensorType) -> None:
         self._container.append(x)
 
     def _aggregate_impl(self) -> NNCFTensor:
-        stacked_val = self._tensor_processor.stack(self._container)
-        aggregated = self._aggregation_fn(stacked_val, axis=self._aggregation_axes, keepdims=self._keepdims)
-        return self._tensor_processor.squeeze(aggregated, 0).tensor
+        # Case when all registered tensors have identical shape
+        if all(self._container[0].shape == x.shape for x in self._container):
```

Comment:
```suggestion
        if all(self._container[0].shape == x.shape for x in self._container[1:]):
```

And please add empty line before comments of cases,

Reply:
Empty lines before each case comments are added